### PR TITLE
RST syntax and tree fixes

### PR DESCRIPTION
**Pull request by rs2**
Remaining Sphinx warnings:
```
WARNING: error while formatting arguments for distributed.diagnostics.progress: module 'distributed.diagnostics.progress' has no attribute '__mro__'
WARNING: html_static_path entry '~/code/distributed/docs/source/_static' does not exist
```

### CODE REVIEW

File: docs/source/ec2.rst [hunk 1]
```diff
@@ -1 +1,4 @@
+Cloud Deployment
+=================
+
 See `Dask's cloud deployment documentation <https://dask.pydata.org/en/latest/setup/cloud.html>`_ for up-to-date documentation for deployment on Amazon's Cloud.
```
Comment (mrocklin) [inline docs/source/ec2.rst]: We no longer publish this file.  It's only here in order to redirect users.  

Probably we should just use a proper redirect.  There is a dictionary of redirects in `source/conf.py` that might be good for this.

File: docs/source/index.rst [hunk 2]
```diff
@@ -124,6 +125,7 @@ Contents
    :caption: Developer Documentation
 
    changelog
+   ec2
```
Comment (mrocklin) [inline docs/source/index.rst]:
```suggestion
```

File: docs/source/worker.rst [hunk 3]
```diff
@@ -239,7 +239,7 @@ Nanny
 Dask workers are by default launched, monitored, and managed by a small Nanny
 process.
 
-.. autoclass:: distributed.worker.Nanny
+.. autoclass:: distributed.nanny.Nanny
```
Comment (mrocklin) [inline docs/source/worker.rst]: Whoops.  Thanks!

**Comment by rs2** [comment]
@mrocklin Thank you, over to you.

**Comment by mrocklin** [comment]
Thanks @rs2 !  Merged in.

**Pull request merged**